plugins/git/plugin.zsh

```diff
@@ -54,7 +54,7 @@ prompt_geometry_git_branch() {
 }
 
 prompt_geometry_git_status() {
-  if test -z "$(git status --porcelain --ignore-submodules)"; then
+  if test -z "$(git status --porcelain --ignore-submodules HEAD)"; then
```

Comment:
The issue here is that using `HEAD` it only checks for tracked files.

Although we can add something specific for untracked files, like `git ls-files --others --exclude-standard` (idk if we can optimize that command further) and use your version for tracked files as an optimization: quickly check if tracked files are dirty, if not take a longer look at untracked files.

Most of the time repos are dirty because of tracked files. Programmers aren't adding new files that often and if they are in a repo which is clean, they're probably going make dirty soon enough. We can go with 2 checks if the total time isn't *much* longer than the current time in the live version. Most of the time, the second check won't even be needed.

Can you measure performance on your end so we can have comparable data from the same sources?